### PR TITLE
Convert module from javac to jdt

### PR DESCRIPTION
**Pull request by datho7561**

### CODE REVIEW

**Comment by datho7561** [comment]
Need to handle the contents of the module still

File: org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacConverter.java [hunk 1]
```diff
@@ -142,6 +143,9 @@ void populateCompilationUnit(CompilationUnit res, JCCompilationUnit javacCompila
 		if (javacCompilationUnit.getPackage() != null) {
 			res.setPackage(convert(javacCompilationUnit.getPackage()));
 		}
+		if (javacCompilationUnit.getModule() != null) {
```
Comment (fbricon) [inline org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacConverter.java:154]: let's say your project is a java module, i.e there's 1 module-info.java and there are N classes, what does javacCompilationUnit.getModule() mean? Is it:
- only module-info.java has a javacCompilationUnit.getModule() != null, in which case It's probably fine as it is,
- each compilation unit of has a javacCompilationUnit.getModule()? in which case do we really need to create N instances of module, 1 per CU, or can one be shared across all of those CUs?

Comment (fbricon) [reply org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacConverter.java:154]: I guess the same question applies for packages

Comment (datho7561) [reply org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacConverter.java:154]: I'm pretty sure that only module-info.java has a module.

**Comment by datho7561** [comment]
@robstryker if you have a chance, you could please take a look at this? Some thing I don't really know how to deal with is when converting the modifiers `transient` and `static`, there is no corresponding javac node. What have you been doing in these cases? We might be able to perform string manipulation to calculate the offsets.

**Comment by robstryker** [comment]
Hey @datho7561 

So there were some situations where I had something similar. I've been using the `getJLS2ModifiersFlagsAsStringLength` method to calculate the offsets directly. I've only had to do this in one occasion so far, and it was only in the case of JLS 2 where a constructor had a void return but the DOM tree / test cases still expected a node to be there.  Very niche case. 

```
	retType = this.ast.newPrimitiveType(convert(TypeKind.VOID));
	// // TODO need to find the right range
	retType.setSourceRange(javac.mods.pos + getJLS2ModifiersFlagsAsStringLength(javac.mods.flags), 0); 
```

**Comment by datho7561** [comment]
Thanks, I'll look into using this

**Pull request merged**